apps/insights/src/components/PriceFeed/Chart/chart.tsx

```diff
@@ -0,0 +1,490 @@
+          const currentHistoricalConfidenceHighData =
+            chartRef.current.confidenceHigh.data().filter((d) => isLineData(d));
+          const currentHistoricalConfidenceLowData =
+            chartRef.current.confidenceLow.data().filter((d) => isLineData(d));
+
+          const newHistoricalPriceData = data.map((d) => ({
+            time: Number(d.timestamp) as UTCTimestamp,
+            value: d.price,
+          }));
+          const newHistoricalConfidenceHighData = data.map((d) => ({
+            time: Number(d.timestamp) as UTCTimestamp,
+            value: d.price + d.confidence,
+          }));
+          const newHistoricalConfidenceLowData = data.map((d) => ({
+            time: Number(d.timestamp) as UTCTimestamp,
+            value: d.price - d.confidence,
+          }));
+
+          // Combine the current and new historical price data
+          const mergedPriceData = mergeData(
+            currentHistoricalPriceData,
+            newHistoricalPriceData,
+          );
+          const mergedConfidenceHighData = mergeData(
+            currentHistoricalConfidenceHighData,
+            newHistoricalConfidenceHighData,
+          );
+          const mergedConfidenceLowData = mergeData(
+            currentHistoricalConfidenceLowData,
+            newHistoricalConfidenceLowData,
+          );
+
+          // Set the new historical price data and (maybe) reset the visible range
+          chartRef.current.price.setData(mergedPriceData);
+          chartRef.current.confidenceHigh.setData(mergedConfidenceHighData);
+          chartRef.current.confidenceLow.setData(mergedConfidenceLowData);
+          maybeResetVisibleRange();
+          didLoadInitialData.current = true;
+        })
+        .catch((error: unknown) => {
+          if (error instanceof Error && error.name === "AbortError") {
+            return;
+          }
+          logger.error("Error fetching historical prices", error);
+        })
+        .finally(() => {
+          isBackfilling.current = false;
+        });
+    },
+    [symbol, chartRef, logger],
+  );
+
+  useMountEffect(() => {
+    const chartElem = chartContainerRef.current;
+    if (chartElem === null) {
+      throw new Error("Chart element was null on mount");
+    }
+    if (chartRef.current) {
+      chartRef.current.chart.remove();
+    }
+    const chart = createChart(chartElem, {
+      layout: {
+        attributionLogo: false,
+        background: { color: "transparent" },
+      },
+      grid: {
+        horzLines: {
+          color: "transparent",
+        },
+        vertLines: {
+          color: "transparent",
+        },
+      },
+      timeScale: {
+        timeVisible: true,
+        secondsVisible: true,
+      },
+      localization: {
+        priceFormatter: priceFormatter.format,
+      },
+    });
+
+    const confidenceHigh = chart.addSeries(AreaSeries, confidenceConfig);
+    const confidenceLow = chart.addSeries(AreaSeries, confidenceConfig);
```

Comment:
Since you moved to using an AreaSeries and don't need to draw the two lines, could you compress this into a single series centered on the price instead of a series above and one below?

Reply:
i don't think we can do this if we want the purple background between the lines, but i can have another look